openedx/openedx-platform
# [BD-3] [TNL-7617] [BB-2969] Add new filters & data

We modify the discussions app to create new filters (author[Only posts by current user], post_type[Discussion or Question], flagged[Only threads that are flagged by any user], following[TBD]) as well as new data in CommentSerializer[abuse_flagged_any_user=Whether comment was flagged by any user] and ThreadSerializer[count_flags=Total number of flags on all comments in the thread]

**JIRA tickets**: BLENDED-583 BLENDED-584 TNL-7617

**Dependencies**: https://github.com/edx/cs_comments_service/pull/328

**Sandbox URL**: TBD - sandbox is being provisioned.

**Merge deadline**: None

**Testing instructions**:

1. Checkout this PR
2. Try setting author, post_type or flagged and see if the filtering works properly
3. Verify that all threads and comments always have count_flags and abuse_flagged_any_user respectively

**Author notes and concerns**:

1. I still need to update tests and spec files
2. We were not able to add functionality to topics since the discussions rest api grabs topic data from modulestore instead of cs_comments_service

**Reviewers**
